### PR TITLE
add a new tracer of coarsepm

### PR DESCRIPTION
**Pull request by haiqinli**
1. Add a new tracer of coarsepm, which is mainly from the coarse dust emission.
2. A bug fix of dust emission from Barry.
3. An update of smoke/dust direct and indirect feedback.
4. An update of plumerise.

The regression test was successful on Hera with Intel compiler.(/scratch2/BMC/acomp/Haiqin.Li/Commit/RRFS_dev/coarsepm.ufs-weather-model/tests/RegressionTests_hera.intel.log)

### CODE REVIEW

File: physics/GFS_rrtmg_pre.F90 [hunk 1]
```diff
@@ -608,10 +608,10 @@ subroutine GFS_rrtmg_pre_run (im, levs, lm, lmk, lmp, n_var_lndp,        &
        if (aero_dir_fdb) then
          do k=1,lmk
            do i=1,im
-             aer_nm(i,k,1 )=aer_nm(i,k,1 )+qgrs(i,k,ntdust )*0.11
-             aer_nm(i,k,2 )=aer_nm(i,k,1 )+qgrs(i,k,ntdust )*0.89
-             aer_nm(i,k,12)=aer_nm(i,k,12)+qgrs(i,k,ntsmoke)*0.01
-             aer_nm(i,k,14)=aer_nm(i,k,14)+qgrs(i,k,ntsmoke)*0.99
+             aer_nm(i,k,1 )=aer_nm(i,k,1 )+qgrs(i,k,ntdust )*1.e-9*0.11
```
Comment (zhanglikate) [inline physics/GFS_rrtmg_pre.F90]: Please use consistent way as GFS to add the tracers from chemical model for radiative forcing: https://github.com/NCAR/ccpp-physics/pull/910/commits/a02422c45b06f46cf0a61ccfcd3083666b2918d0 , by this using way, the GFS will not be impact by the order of chemical tracers, which would be safe for long-term use.

**Review by zhanglikate — CHANGES_REQUESTED** [review]

File: physics/GFS_rrtmg_pre.F90 (hunk 1 above)
Comment (haiqinli) [reply physics/GFS_rrtmg_pre.F90]: @zhanglikate Thanks for your suggestions. Yes, our way is similar to that in https://github.com/NCAR/ccpp-physics/commit/a02422c45b06f46cf0a61ccfcd3083666b2918d0. We treat part of smoke/dust as anomaly, and add on top of the climatology. We will further tune the magnitude in our following test runs.

**Review by tanyasmirnova — APPROVED** [review]

File: physics/mynnedmf_wrapper.meta [hunk 2]
```diff
@@ -1398,7 +1398,7 @@
   standard_name = dry_deposition_velocity_mynn_pbl_transport
   long_name = dry deposition velocity by mynn pbl transport
   units = m s-1
-  dimensions = (horizontal_loop_extent,2)
+  dimensions = (horizontal_loop_extent,3)
   type = real
```
Comment (joeolson42) [inline physics/mynnedmf_wrapper.meta:1402]: Is there any chance you can turn these hard-coded values into a derived variable?

**Review by joeolson42 — APPROVED** [review]
You may get questioned at some point about using "rrfs" in the name of your modules and namelist options. Using "gfs" is also now being discouraged, so at some point, the repository managers will ask you to remove specific applications names from your code. Just a heads up. I will approve the code as is, but suggest changing the hard-coded number to variables at some point.

**Comment by haiqinli** [comment]
@joeolson42 Thanks for your approval and suggestions. We will resolve the hard-coded number of variables soon.

**Review by mdtoyNOAA — APPROVED** [review]

**Review by zhanglikate — APPROVED** [review]
I would suggest to merge with the most recent version of ccpp-physics at the earliest, some implementations are already or similar in the recent version of ccpp-physics, better to follow the same way for new implementations.

**Pull request merged**